src/main/java/co/cask/gcs/sink/GCSParquetBatchSink.java

```diff
@@ -61,68 +57,35 @@ public void initialize(BatchRuntimeContext context) throws Exception {
   }
 
   @Override
-  protected OutputFormatProvider createOutputFormatProvider(BatchSinkContext context) {
-    return new GCSParquetOutputFormatProvider(config, context);
+  public void transform(StructuredRecord input,
+                        Emitter<KeyValue<Void, GenericRecord>> emitter) throws Exception {
+    emitter.emit(new KeyValue<>(null, recordTransformer.transform(input)));
   }
 
   @Override
-  public void transform(StructuredRecord input,
-                        Emitter<KeyValue<Void, GenericRecord>> emitter) throws Exception {
-    emitter.emit(new KeyValue<Void, GenericRecord>(null, recordTransformer.transform(input)));
+  protected String getOutputFormatClassname() {
+    return AvroParquetOutputFormat.class.getName();
+  }
+
+  @Override
+  protected Map<String, String> getOutputFormatConfig() {
+    Map<String, String> conf = new HashMap<>();
+    conf.put(PARQUET_AVRO_SCHEMA, config.schema);
+    conf.putAll(FileSetUtil.getParquetCompressionConfiguration(config.codec, config.schema, true));
+    return conf;
   }
 
   /**
-   * Configuration for the S3ParquetConfig.
+   * Configuration for the sink.
    */
   public static class GCSParquetSinkConfig extends GCSBatchSinkConfig {
     @Name("schema")
-    @Description("Specifies the schema to used for defining the GCS Dataset.")
+    @Description("The schema of records to write.")
     @Macro
     protected String schema;
 
     @Nullable
-    @Description("Used to specify the compression codec to be used for the final dataset.")
+    @Description("The compression codec to use when writing data. Must be 'none', 'snappy', 'gzip', or 'lzo'.")
```

Comment:
similar comment about supported compression types.

Reply:
this is correct for parquet. avro and parquet support different codecs apparently